src/pages/transaction-signing/components/confirm-send-drawer.tsx

```diff
@@ -85,6 +104,98 @@ const Actions: React.FC<{ transaction: StacksTransaction | null; handleCancel: (
   );
 };
 
+interface TxSettingsProps {
+  transaction: StacksTransaction | null;
+  setTransaction: (transaction: StacksTransaction) => void;
+  amount: number;
+  memo: string;
+  recipient: string;
+}
+
+const TxSettings: React.FC<TxSettingsProps> = ({
+  transaction,
+  setTransaction,
+  amount,
+  memo,
+  recipient,
+}) => {
+  const MAX_LENGTH_BIG_NUMBER = 16;
+  const [nonceOverride, setNonceOverride] = React.useState(
+    transaction?.auth?.spendingCondition?.nonce || 0
+  );
+  const network = useAtomValue(currentStacksNetworkState);
+  const account = useAtomValue(currentAccountState);
+  if (!account) return <></>;
+  const senderKey = account.stxPrivateKey;
+
+  if (!senderKey) return <></>;
+  const stxAddress = account?.address;
+
+  const { selectedAsset } = useSelectedAsset();
+  const balances = useAtomValue(accountBalancesState);
+
+  const handleChange = async (e: React.ChangeEvent<any>) => {
+    if (!account || e.target.value.length >= MAX_LENGTH_BIG_NUMBER) return;
+    let nonce;
+    try {
+      nonce = new BN(e.target.value, 10).toNumber();
+    } catch (e) {
+      // BN will throw an exception if not an integer, we keep the last value in that case
+      return;
+    }
+
+    setNonceOverride(e.target.value);
+    if (!selectedAsset) return;
+    const bnAmount = stxToMicroStx(amount).toString();
+    let tx;
+
+    // Build the transaction again since the nonce has changed
+    if (selectedAsset.type === 'stx') {
+      let txData = {
+        txType: TransactionTypes.STXTransfer,
+        amount: bnAmount,
+        memo,
+        recipient,
+        network,
+      } as STXTransferPayload;
+      tx = await generateSTXTransferTx({ txData, senderKey, nonce });
+    } else {
+      let txData = {
+        amount,
+        balances,
+        memo,
+        network,
+        nonce,
+        recipient,
+        selectedAsset,
+        senderKey,
+        stxAddress,
+      };
+      tx = await buildAssetTransferTx(txData);
+    }
```

Comment:
Is it possible to use the `StacksTransaction.setNonce()` method here, rather than rebuilding the tx ourselves?

Reply:
The issue is that the transaction we get in the preview is already signed, so if you change the nonce, the signature is not valid anymore and will be rejected by the api.

Reply:
I see. This functionality will be pretty difficult to implement when we support Ledger, then. The user'd have to go through the tx signing flow again?

Reply:
The signed transaction is a derived piece of state -- meaning you can change any of the dependencies (nonce, network, account) and it will automatically update. I'd avoid recommending using any of the built in methods in the `StacksTransaction` class to modify its values, only the getters imo.

Reply:
I think this 'edit nonce' is not needed for the typical edger user. I supposed this is mainly for developer/advanced users.